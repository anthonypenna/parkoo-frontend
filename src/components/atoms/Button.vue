<template>
  <button v-bind="$attrs" :class="`button button--${theme}`">
    <slot />
  </button>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'Button',

  props: {
    theme: {
      default: 'secondary',
      type: String
    }
  }
})
</script>

<style lang="scss" scoped>
.button {
  max-width: 295px;
  width: 100%;
  padding: 14px 0;
  border: none;
  border-radius: 5px;
  font-family: inherit;
  font-size: 16px;
  font-weight: bold;

  &--primary {
    background: #ff9979;
    color: #fff;
  }

  &--secondary {
    background: #f1f1f1;
    color: #000;
  }
}
</style>
