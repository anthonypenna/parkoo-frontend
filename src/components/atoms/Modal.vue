<template>
  <Fade>
    <div class="modal">
      <slot />
    </div>
  </Fade>
</template>

<script lang="ts">
import { Map } from "mapbox-gl";
import Vue from "vue";
import Fade from "@/components/atoms/Fade.vue";
import { mapGetters } from "vuex";

export default Vue.extend({
  name: "Modal",

  components: { Fade },

  computed: {
    ...mapGetters("mapbox", ["map"]),
  },

  mounted() {
    const map = this.map as Map;
    map.dragPan.disable();
    map.scrollZoom.disable();
  },

  destroyed() {
    const map = this.map as Map;
    map.dragPan.enable();
    map.scrollZoom.enable();
  },
});
</script>

<style lang="scss" scoped>
.modal {
  position: fixed;
  top: 50%;
  left: 50%;
  max-width: 560px;
  width: calc(100% - (56px + 2rem));
  padding: 32px 28px 16px;
  background: #fff;
  border-radius: 15px;
  box-shadow: 0 4px 16px rgba($color: #000000, $alpha: 0.1);
  transform: translate(-50%, -50%);
  z-index: 10;
}
</style>
