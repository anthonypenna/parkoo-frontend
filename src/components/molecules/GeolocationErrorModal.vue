
<template>
  <Modal>
    <h2>Oh oh!</h2>
    <p>
      Parkoo needs access to your <strong>location</strong> in order to work
      correctly.
    </p>
    <p>
      Please check your <strong>settings</strong>, refresh the page and try
      again
    </p>
  </Modal>
</template>

<script lang="ts">
import Vue from "vue";
import Modal from "@/components/atoms/Modal.vue";

export default Vue.extend({
  name: "GeolocationErrorModal",
  components: { Modal },
});
</script>
