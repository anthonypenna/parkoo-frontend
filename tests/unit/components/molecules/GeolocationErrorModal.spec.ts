import { mount } from '@vue/test-utils'
import GeolocationErrorModal from '@/components/molecules/GeolocationErrorModal.vue'
import Button from '@/components/atoms/Button.vue'
import Modal from '@/components/atoms/Modal.vue'

describe('<GeolocationErrorModal />', () => {
  it('should render correctly', () => {
    const wrapper = mount(GeolocationErrorModal, { stubs: { Modal: true } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('when clicking the try again button', () => {
    it('should emit a try again event', () => {
      const wrapper = mount(GeolocationErrorModal, { stubs: { Modal: true } })
      wrapper.findComponent(Button).trigger('click')
      expect(wrapper.emitted().tryagain).toBeTruthy()
    })
  })
})
