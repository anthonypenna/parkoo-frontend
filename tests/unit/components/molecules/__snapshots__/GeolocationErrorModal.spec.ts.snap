// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeolocationErrorModal /> should render correctly 1`] = `
<modal-stub>
  <h2>Oops!</h2>
  <p>
    Parkoo needs access to your location <br>in order to work correctly.
  </p>
  <p>Please check your settings and try again</p> <button class="button button--secondary">Try again</button>
</modal-stub>
`;
